### PR TITLE
Table model (list of lists) for QTableView

### PR DESCRIPTION
**Pull request by kernc**
Obsoletes https://github.com/biolab/orange3/pull/1208

### CODE REVIEW

**Comment by astaric** [comment]
Could you add some tests for the new model?

**Comment by kernc** [comment]
Some tests added.

File: Orange/widgets/utils/itemmodels.py [hunk 1]
```diff
@@ -49,15 +47,156 @@ def signal_blocking(obj):
...
+        self._table = []
+        self._headers = {}
+        if sequence is not None:
+            self.extend(sequence)
+
+    def rowCount(self, parent=QModelIndex()):
+        return 0 if parent.isValid() else len(self)
+
+    def columnCount(self, parent=QModelIndex()):
+        return 0 if parent.isValid() or not len(self) else max((len(row) for row in self._table), default=0)
```
Comment (janezd) [inline Orange/widgets/utils/itemmodels.py]: Either `not len(self)` or `default=0` is unnecessary, isn't it?

File: Orange/widgets/utils/itemmodels.py [hunk 2]
```diff
@@ -49,15 +47,156 @@ def signal_blocking(obj):
...
+            try:
+                yield
+            finally:
+                self.endInsertColumns()
+        else:
+            yield
+
+    def extend(self, rows):
+        i, rows = len(self), list(rows)
+        self.beginInsertRows(QModelIndex(), i - 1, i - 1 + len(rows) - 1)
```
Comment (janezd) [inline Orange/widgets/utils/itemmodels.py]: Remove `-1`

**Pull request merged**